Support: javascript-node 24

Marukome0743 opened the pull request:
## Summary

Introduce Node.js 24 on both Debian Bookworm and Bullseye, make it the default variant, and update documentation and tag configurations accordingly.

New Features:
- Add support for Node.js 24 variants (`24-bookworm` and `24-bullseye`) in the JavaScript Node image

Enhancements:
- Promote `24-bookworm` to the default (`latest`) variant
- Update variant tag mappings and supported architectures for all Debian variants
- Refresh README to include new 24 variants and adjust sample image tag examples
- Bump the default `VARIANT` in the devcontainer Dockerfile to `24-bookworm`

haltcase commented:
Just curious, is this blocked by anything in particular, such as v24 becoming LTS? Or is it simply waiting for availability to review?

AlvaroRausell reviewed (changes requested):
As a reply to https://github.com/devcontainers/images/pull/1423#discussion_r2143750487
_(Disclaimer: it is late and I am relatively new to the org)_
It seems like there is a need for the JS version to be published _before_ we can move onto the typescript variant. What I will ask you to do is: remove all Typescript changes, we will publish the JS ones and then make another PR for TS which we can deal with once the former has been merged and released (I will do by EOD Monday).

Marukome0743 commented:
Thank you for a nice review!
As you say, I removed the typescript changes from this PR.
In addition, I have updated the PR title and description.

On the other hand, I will make the another typescript 24 PR.

Marukome0743 commented on src/javascript-node/README.md:
```diff
@@ -20,18 +20,19 @@
-- `mcr.microsoft.com/devcontainers/typescript-node:2.0.0-22` (or `2.0.0-22-bookworm`, `2.0.0-22-bullseye`)
+- `mcr.microsoft.com/devcontainers/typescript-node:2-24` (or `2-24-bookworm`, `2-24-bullseye`)
+- `mcr.microsoft.com/devcontainers/typescript-node:2.0-24` (or `2.0-24-bookworm`, `2.0-24-bullseye`)
+- `mcr.microsoft.com/devcontainers/typescript-node:2.0.0-24` (or `2.0.0-24-bookworm`, `2.0.0-24-bullseye`)
```
Oh, I missed... Thank you for nice check!

Marukome0743 commented:
![YayYeahGIF](https://github.com/user-attachments/assets/83723b27-4b7e-483a-b1cd-5e5ad86010dd)
